src/main/java/ch/uzh/ifi/hase/soprafs23/controller/GroupController.java

```diff
@@ -224,6 +224,39 @@ public List<UserGetDTO> getGroupMembersById(@PathVariable Long groupId, HttpServ
         return userGetDTOs;
     }
     
+    @GetMapping("/groups/{groupId}/guests")
+    @ResponseStatus(HttpStatus.OK) // 200
+    @ResponseBody
+    public List<UserGetDTO> getGroupGuestsById(@PathVariable Long groupId, HttpServletRequest request) {
+        // 404 - group not found
+        Group group = groupService.getGroupById(groupId);
+
+        // 401 - not authorized
+        Long tokenId = userService.getUseridByToken(request.getHeader("X-Token"));
+        if(tokenId.equals(0L)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, String.format("You are not authorized."));
+        }
+
+        // get the guests of the group
+        List<Long> guestIds = groupService.getAllGuestIdsOfGroup(group);
+
+        // 204 - no open invitations
```

Comment:
is this really checking the inventations? I guess its checking only whether people are in the group?

Reply:
do you mean invitations? no it doesn't check those at all it just returns the people actually in the group, from our talk I think that is what is needed

Reply:
Yeah, spelling mistake. I was confused by the comment "no open INVItations"